src/internal.c

```diff
@@ -15377,6 +15377,12 @@ int ProcessReply(WOLFSSL* ssl)
             ret = GetRecordHeader(ssl, ssl->buffers.inputBuffer.buffer,
                                        &ssl->buffers.inputBuffer.idx,
                                        &ssl->curRL, &ssl->curSize);
+
+            if (ssl->curSize == 0 && ssl->curRL.type != application_data) {
```

Comment:
I wonder if we should put this in `GetRecordHeader` instead of here after the call site.

Reply:
Could be placed at the end with the other `LENGTH_ERROR` checks.